### ISSUE TITLE
Open a window for debug adapter

### ISSUE THREAD

**Issue by meng-jpg** [eclipse-theia/theia#12561]
### Bug Description:
<img width="1535" alt="截屏2023-05-25 11 03 51" src="https://github.com/eclipse-theia/theia/assets/74631407/6f961d83-da8c-41c7-bd70-f638d78e56de">

### Steps to Reproduce:

1.start theia electron on windows
2.start debug with plugin `cpp-tools`
3.A window of the debug adapter will be opened.

### Additional Information

- Operating System: windows 10 & windows 11
- Theia Version: 1.37.0

**Comment by meng-jpg** [comment]
cpp-tools version: 1.8.4
launch.json:
```
{
        "name": "Debug cpptools",
        "type": "cppdbg",
        "request": "launch",
        "externalConsole": true,
        "program": "C:\\Users\\tptuser\\ZStudio\\TerapinesSolution\\project.exe",
        "args": [],
        "cwd": "C:\\Users\\tptuser\\ZStudio\\TerapinesSolution"
    }
```

**Comment by vince-fugnitto** [comment]
@meng-jpg thank you for reporting the issue, can you share the example workspace you tested against and confirm what the behavior is in vscode? Also note that `vscode-cpptools` is proprietary and cannot legally be used by non vscode products.

**Comment by meng-jpg** [comment]
It's a simply helloworld.c:

```c
#include <stdio.h>

int main() {
    printf("hello\n");
    return 0;
}
```

We compiled the helloworld.c via gnu-gcc and debugged it via the cpptools extension with the launch.json above.

> Also note that `vscode-cpptools` is [proprietary](https://marketplace.visualstudio.com/items/ms-vscode.cpptools/license) and cannot legally be used by non vscode products.

Does it mean that we cannot provide the cpptools to our customers in our IDE as a builtin extension? In that condition, is there any other c/c++ extensions available since we develop a debugger adapter based on the lldb-vscode extension but it's only a debugger and dose not have some intelligent functionalities.

**Comment by msujew** [comment]
> Does it mean that we cannot provide the cpptools to our customers in our IDE as a builtin extension?

Exactly, it cannot be (legally) used by any product outside of official microsoft products. 

> In that condition, is there any other c/c++ extensions available since we develop a debugger adapter based on the lldb-vscode extension but it's only a debugger and dose not have some intelligent functionalities.

Developers building Theia apps usually include the [clangd](https://open-vsx.org/extension/llvm-vs-code-extensions/vscode-clangd) extension instead for C/C++ support. For debugging, you can use the [cdt-gdb-debugger](https://open-vsx.org/extension/eclipse-cdt/cdt-gdb-vscode) extension.

**Comment by meng-jpg** [comment]
When we debugging in vscode via the same version of the cpptools:
<img width="1225" alt="截屏2023-05-26 17 01 02" src="https://github.com/eclipse-theia/theia/assets/74631407/c446ac4c-e825-4ffa-8c79-8a102f061872">
It seems alright.
